packages/syft/src/syft/core/node/common/node_service/generic_payload/syft_message.py

```diff
@@ -0,0 +1,103 @@
+# stdlib
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+# third party
+from nacl.signing import VerifyKey
+from pydantic import BaseModel
+
+# relative
+from .....common.message import ImmediateSyftMessage
+from .....common.message import SignedMessage
+from .....common.uid import UID
+from .....io.address import Address
+from ....abstract.node_service_interface import NodeServiceInterface
+from ....common.exceptions import AuthorizationError
+
+
+# Inner Payload message using Pydantic.
+class Payload(BaseModel):
+    class Config:
+        orm_mode = True
+
+
+class RequestPayload(Payload):
+    pass
+
+
+class ReplyPayload(Payload):
+    pass
+
+
+class NewSyftMessage(ImmediateSyftMessage):
+    """A base class from which all message classes should inherit.
+
+    Note:
+        This will eventually replace the old `SyftMessage` class.
+    """
+
+    __attr_allowlist__ = ["id", "address", "reply_to", "reply", "msg_id", "kwargs"]
+
+    signed_type = SignedMessage
+    request_payload_type = RequestPayload
+    reply_payload_type = ReplyPayload
+
+    def __init__(
+        self,
+        address: Address,
+        kwargs: Optional[Dict[str, Any]] = None,
+        msg_id: Optional[UID] = None,
+        reply_to: Optional[Address] = None,
+        reply: bool = False,
+    ) -> None:
+        super().__init__(address=address, msg_id=msg_id)
+        self.reply_to = reply_to
+        self.reply = reply
+        self.kwargs = kwargs if kwargs else {}
+
+    @property
+    def payload(self) -> Payload:
+        kwargs_dict = {}
+
+        if hasattr(self.kwargs, "upcast"):
+            kwargs_dict = self.kwargs.upcast()  # type: ignore
+        else:
+            kwargs_dict = self.kwargs  # type: ignore
+
+        # If it's not a reply message then load kwargs as a proper request payload.
+        if not self.reply:
+            return self.request_payload_type(**kwargs_dict)
```

Comment:
What happens here if the **splat doesn't match the pydantic model, do we raise an Exception?

Reply:
Now, it captures the `PydanticValidationError` exception and raises a Pysyft/PyGrid custom exception _(`BadPayloadException`)_

Reply:
Fixed by: 
[6a48881](https://github.com/OpenMined/PySyft/pull/6274/commits/6a4888162114da735aa75930cba4973703c5aa75)